tests/tools/onert_train/src/nnfw_util.cc

```diff
@@ -46,4 +49,71 @@ uint64_t bufsize_for(const nnfw_tensorinfo *ti)
   return elmsize[ti->dtype] * num_elems(ti);
 }
 
+std::ostream &operator<<(std::ostream &os, const NNFW_TRAIN_OPTIMIZER &opt)
+{
+  switch (opt)
+  {
+    case NNFW_TRAIN_OPTIMIZER_ADAM:
+      os << "adam";
+      break;
+    case NNFW_TRAIN_OPTIMIZER_SGD:
+      os << "sgd";
+      break;
+    default:
+      os << "unsupported optimzier";
+  }
+  return os;
+}
+
+std::ostream &operator<<(std::ostream &os, const NNFW_TRAIN_LOSS &loss)
+{
+  switch (loss)
+  {
+    case NNFW_TRAIN_LOSS_MEAN_SQUARED_ERROR:
+      os << "mean squared error";
+      break;
+    case NNFW_TRAIN_LOSS_CATEGORICAL_CROSSENTROPY:
+      os << "categorical crossentropy";
+      break;
+    default:
+      os << "unsupported loss";
+  }
+  return os;
+}
+
+std::ostream &operator<<(std::ostream &os, const NNFW_TRAIN_LOSS_REDUCTION &loss_reduction)
+{
+  switch (loss_reduction)
+  {
+    case NNFW_TRAIN_LOSS_REDUCTION_INVALID:
+      os << "invalid (use default)";
+      break;
+    case NNFW_TRAIN_LOSS_REDUCTION_SUM_OVER_BATCH_SIZE:
+      os << "sum over batch size";
+      break;
+    case NNFW_TRAIN_LOSS_REDUCTION_SUM:
+      os << "sum";
+      break;
+    default:
+      os << "unsupported reduction type";
+      break;
+  }
+  return os;
+}
+
+std::ostream &operator<<(std::ostream &os, const nnfw_loss_info &loss_info)
+{
+  os << "{loss= " << loss_info.loss << ", type= " << loss_info.reduction_type << "}";
```

Comment:
```suggestion
  os << "{loss= " << loss_info.loss << ", reduction_type= " << loss_info.reduction_type << "}";
```

Reply:
updated

`reduction_type` is a bit long to display, I rather choose `reduction`.